tests/foreman/api/test_hostgroup.py

```diff
@@ -148,6 +148,64 @@ def test_bz_1107708(self):
         self.assertEqual(host_attrs['all_puppetclasses'][0]['name'], 'ntp')
 
 
+@skip_if_bug_open('bugzilla', 1235377)
+class MissingAttrTestCase(APITestCase):
+    """Tests to see if the server returns the attributes it should."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a ``HostGroup`` as ``cls.host_group``."""
+        cls.host_group = entities.HostGroup().create()
+
+    def test_get_content_source(self):
+        """@Test: Read a host group. Inspect the server's response.
+
+        @Assert: The response contains either a ``content_source`` or
+        ``content_source_id`` attribute.
+
+        @Feature: HostGroup
+
+        """
+        wanted_attrs = ('content_source', 'content_source_id')
```

Comment:
> The solution you submitted was to call `read_json()` in the `SetUp` method and then hard-coding both variations of the attribute you were looking for and asserting they were present. What I missed was that you also wanted to avoid hard-coding those attributes and instead use `_get_entity_id` (please correct me if I'm mistaken).

That's exactly right. :smile:

> I can appreciate the beauty of taking advantage of this method even if it is a private method.

I really like the prospect of dropping the hard-coded name variations and error messages. I can add a brief explanation of the `_get_entity_id` method into the class docstring. That might help out folks who are confused by the tests.